modules/pam_motd/pam_motd.c

```diff
@@ -166,9 +166,24 @@ static int compare_strings(const void *a, const void *b)
     }
 }
 
+static const char* motddir;
+
 static int filter_dirents(const struct dirent *d)
 {
-    return (d->d_type == DT_REG || d->d_type == DT_LNK);
```

Comment:
This will punish every filesystem that already returns filetype here. Fallback is needed only if `d_type == DT_UNKNOWN`.

Reply:
My thinking was that it will be at most 2-3 stat() calls, in a process that isn't performance critical. Keep the code simple and small.  I'll fix it.

Reply:
Please use
```
if (d->d_type != DT_UNKNOWN)
    return d->d_type == DT_REG || d->d_type == DT_LNK;
```

Reply:
Thanks for the feedback. Any comments on using this instead:
```
static int filter_dirents(const struct dirent *d)
{ 
    struct stat s;
    char *fullpath;
    int rc;

    switch( d->d_type )
    {
        case DT_REG:
        case DT_LNK:
            return 1;      /* regular files and symlinks are good         */
        case DT_UNKNOWN:   /* with unknown, we have a bit more work to do */
            break;
        case DT_DIR:       /* We don't want directories   */
        default:           /* nor anything else           */
            return 0;
    }

```
followed by the code that does lstat() etc.

Reply:
Yes, this looks better.
btw, please follow the style of this module, e.g.
```
switch (d->d_type) {
```

Reply:
yup. muscle memory.